### PR TITLE
feat: add support for dnsaddr resolving in browser

### PR DESCRIPTION
**Pull request by zvolin**

### CODE REVIEW

File: node-wasm/src/utils.rs [hunk 1]
```diff
@@ -173,6 +177,97 @@ pub(crate) fn get_crypto() -> Result<Crypto, Error> {
...
+    }) else {
+        // not a dnsaddr multiaddr
+        return Ok(vec![ma]);
+    };
+
+    let mut opts = RequestInit::new();
+    opts.method("GET");
+    opts.mode(RequestMode::Cors);
+
+    let url = format!("https://1.1.1.1/dns-query?type=txt&name=_dnsaddr.{dnsaddr}");
```
[the remark this answers is not in the document]
Comment (zvolin) [reply node-wasm/src/utils.rs:243]: Just to note, we also explicitly setup cloudflare for dns in native node

[the remark this answers is not in the document]
Comment (zvolin) [reply node-wasm/src/utils.rs]: done

File: node-wasm/src/utils.rs [hunk 2]
```diff
@@ -173,6 +178,100 @@ pub(crate) fn get_crypto() -> Result<Crypto, Error> {
...
+    let Some(dnsaddr) = ma.iter().find_map(|protocol| {
+        if let Protocol::Dnsaddr(addr) = protocol {
+            Some(addr)
+        } else {
+            None
+        }
+    }) else {
+        // not a dnsaddr multiaddr
+        return Ok(vec![ma]);
+    };
```
Comment (oblique) [inline node-wasm/src/utils.rs:240]: This `let-else` is hard to read. Mainly because `else` comes after few lines.

File: node-wasm/src/utils.rs [hunk 3]
```diff
@@ -173,6 +178,100 @@ pub(crate) fn get_crypto() -> Result<Crypto, Error> {
...
+            let Ok(data) = serde_json::from_str::<String>(&entry.data) else {
+                continue;
+            };
+            let Some((_, ma)) = data.split_once('=') else {
+                continue;
+            };
+            let Ok(ma) = ma.parse() else {
+                continue;
+            };
+            resolved_addrs.push(ma);
```
Comment (oblique) [inline node-wasm/src/utils.rs:272]: We must accept only multiaddresses that have the same p2p key as the original one.

Comment (oblique) [reply node-wasm/src/utils.rs:272]: We may need recursion too ([link](https://github.com/multiformats/multiaddr/blob/master/protocols/DNSADDR.md#suffix-matching)), but we can implement it in another PR.

Comment (zvolin) [reply node-wasm/src/utils.rs]: done + left a todo for recursion

**Review by fl0rek — APPROVED** [review]

**Review by oblique — APPROVED** [review]
LGTM

**Pull request merged**